src/libYARP_OS/src/Network.cpp

```diff
@@ -871,6 +872,40 @@ bool NetworkBase::getConnectionQos(const ConstString& src, const ConstString& de
     return true;
 }
 
+bool NetworkBase::isValidPortName(const ConstString& portName)
+{
+    if (portName.empty())
+    {
+        fprintf(stderr, "Failure: invalid port name, the string is empty\n");
```

Comment:
I would remove the `fprintf()`s, and leave the output to the caller.
This method should just return true or false, calling it with an invalid port name is not necessarily a "failure".